### PR TITLE
[search] Latvian search synonyms for categories.txt

### PR DESCRIPTION
**Pull request by biodranik**
Please review all changes. Note that they should match types_strings.txt translations.

### CODE REVIEW

**Review by ProgramminCat — APPROVED** [review]

**Comment by pastk** [comment]
Tests are failing because `lv` is not enabled somewhere yet.

**Comment by pastk** [comment]
Please add `lv:cietums` for amenity-prison from #9475

**Comment by pastk** [comment]
Also translations are not generated off `strings.txt` yet, see https://github.com/organicmaps/organicmaps/pull/9211#issuecomment-2420769075

File: data/categories.txt [hunk 1]
```diff
@@ -16391,6 +16842,7 @@ et:Jooksurada
 fr:Piste de course
 it:Pista da corsa
 ja:トラック
+lv:Skriešanas celiņš
```
Comment (soshial) [inline data/categories.txt:16845]: also "Skrējceļš"

File: data/categories.txt [hunk 2]
```diff
@@ -16325,6 +16774,7 @@ id:Jam tangan
 it:Orologeria
 ja:時計
 ko:시계
+lv:Pulksteņi
```
Comment (soshial) [inline data/categories.txt:16777]: also "Pulksteņu darbnīca"

File: data/categories.txt [hunk 3]
```diff
@@ -16156,7 +16601,7 @@ de:Nähzubehör
 el:Είδη Ραπτικής
 es:Materiales de costura|Mercería
 et:Õmblustarvikud
-eu:Josteko hornigaiak
+eu:Josteko hornigaiakt
```
Comment (soshial) [inline data/categories.txt:16604]: Is this a typo? Or do you speak Basque language, @ttrasuns ?

File: data/categories.txt [hunk 4]
```diff
@@ -15732,6 +16166,7 @@ id:Toko Tas
 it:Negozio di borse
 ja:バッグストア
 ko:가방 판매점
+lv:Maisi
```
Comment (soshial) [inline data/categories.txt:16169]: "Somas"

File: data/categories.txt [hunk 5]
```diff
@@ -14495,6 +14891,7 @@ id:4Defibrilator
 it:4Defibrillatore
 ja:除細動器
 ko:제세동기
+lv:4Defibrilators
```
Comment (soshial) [inline data/categories.txt:14894]: also "Automatizētais ārējais defibrilators"

File: data/categories.txt [hunk 6]
```diff
@@ -14113,6 +14499,7 @@ id:Sekolah bahasa
 it:Scuola di lingue
 ja:語学学校
 ko:언어 학교
+lv:Valodu skola
```
Comment (soshial) [inline data/categories.txt:14502]: also "Svešvalodas"

File: data/categories.txt [hunk 7]
```diff
@@ -13922,6 +14303,7 @@ id:5Apartemen
 it:Residence
 ja:アパート
 ko:아파트
+lv:Apartamenti
```
Comment (soshial) [inline data/categories.txt:14306]: also "Rezidences"

File: data/categories.txt [hunk 8]
```diff
@@ -13847,6 +14226,7 @@ id:Kolam Renang
 it:Centro natatorio
 ja:スイミングセンター
 ko:수영 센터|수영장
+lv:Peldēšana
```
Comment (soshial) [inline data/categories.txt:14229]: also "Peldbaseins|Baseins|Peldvieta"

File: data/categories.txt [hunk 9]
```diff
@@ -13737,6 +14113,7 @@ id:Rak Buku|Bursa Buku
 it:Libreria|Scambio libri
 ja:書棚|図書交換
 ko:책 교환|도서 교환|책 저장소
+lv:Grāmatu maiņa
```
Comment (soshial) [inline data/categories.txt:14116]: also "Grāmatplaukts"

File: data/categories.txt [hunk 10]
```diff
@@ -13443,6 +13811,7 @@ id:Taman air
 it:Parco acquatico
 ja:ウォーターパーク
 ko:워터파크|물놀이 공원
+lv:Ūdens atrakciju parks
```
Comment (soshial) [inline data/categories.txt:13814]: Also "akvaparks"

File: data/categories.txt [hunk 11]
```diff
@@ -12745,6 +13093,7 @@ id:Titik penjemputan
 it:Punto di raccolta
 ja:集荷場
 ko:픽업 장소
+lv:Saņemšanas punkts
```
Comment (soshial) [inline data/categories.txt:13096]: Should be "Preču saņemšanas punkts"

File: data/categories.txt [hunk 12]
```diff
@@ -12297,6 +12629,7 @@ id:Kantor wisata
 it:Ufficio turistico
 ja:観光案内所
 ko:관광정보센터|안내소|관광 안내소|여행객 안내소|여행자 안내소
+lv:Tūrisma birojs
```
Comment (soshial) [inline data/categories.txt:12632]: Should be "Tūrisma informācijas centrs|TIC"

File: data/categories.txt [hunk 13]
```diff
@@ -11789,6 +12106,7 @@ id:Dokter Hewan
 it:4Veterinario
 ja:獣医
 ko:수의사|동물병원
+lv:4Veterinārārsts
```
Comment (soshial) [inline data/categories.txt:12109]: also "vetārsts"

File: data/categories.txt [hunk 14]
```diff
@@ -9997,6 +10267,7 @@ id:5Perkemahan
 it:4Campeggio|campeggi|camping
 ja:1キャンプ場|キャンプ|キャンピング|バーベキュー|テント場
 ko:캠프|캠프장|캠핑을|캠핑|캠핑장|캠핑장소
+lv:4Kempings|teltis|U+26FA
```
Comment (soshial) [inline data/categories.txt:10270]: Also "telšu pilsētiņa"

File: data/categories.txt [hunk 15]
```diff
@@ -8634,6 +8869,7 @@ id:Berkuda
 it:Equitazione
 ja:馬術
 ko:승마|마술 스포츠
+lv:Jāšana|jāšanas sports
```
Comment (soshial) [inline data/categories.txt:8872]: also "zirgi"

File: data/categories.txt [hunk 16]
```diff
@@ -7510,13 +7713,15 @@ mr:शिखर
...
 en:Peak
+lv:smaile
 
 natural-saddle|mountain_pass
 en:4Saddle|Pass
 be:4Седлавіна|5Перавал
 et:Sadul
 fr:Col
 it:4Sella
+lv:Pāreja
```
Comment (soshial) [inline data/categories.txt:7724]: also "Kalnu pāreja"

File: data/categories.txt [hunk 17]
```diff
@@ -7510,13 +7713,15 @@ mr:शिखर
 # Add at least one lang not to be skipped.
 natural-peak|@mountain
 en:Peak
+lv:smaile
```
Comment (soshial) [inline data/categories.txt:7716]: also "virsotne"

File: data/categories.txt [hunk 18]
```diff
@@ -6594,7 +6772,7 @@ sv:Vindskydd
 th:ที่อยู่อาศัย
 tr:Barınak
 uk:Бівуачний навіс
-vi:Chổ nương thân
+vi:Chổ nương thân
```
Comment (soshial) [inline data/categories.txt:6775]: Typo?

File: data/categories.txt (hunk 1 above)
Comment (richlv) [reply data/categories.txt:16845]: "Skrejceļš" here

File: data/categories.txt (hunk 5 above)
Comment (biodranik) [reply data/categories.txt:14894]: It is unlikely that someone will type this in the search...

File: data/categories.txt (hunk 11 above)
Comment (biodranik) [reply data/categories.txt:13096]: Does someone type it like this in the search?

File: data/categories.txt (hunk 18 above)
Comment (biodranik) [reply data/categories.txt:6775]: Unicode symbols were properly merged in the editor here )

**Comment by biodranik** [comment]
Thanks to everyone for the review, I've updated the PR. There are several important untranslated files left to publish a release with Latvian support:

- Text-to-speech strings for navigation: [data/strings/sound.txt](https://github.com/organicmaps/organicmaps/blob/master/data/strings/sound.txt)
- A few iOS specific strings: [iphone/plist.txt](https://github.com/organicmaps/organicmaps/blob/master/iphone/plist.txt)
- Names of map features/types (should match those in categories.txt): [data/strings/types_strings.txt](https://github.com/organicmaps/organicmaps/blob/master/data/strings/types_strings.txt)

CC @ttrasuns @organicmaps/translations-lv

File: indexer/categories_holder.hpp [hunk 19]
```diff
@@ -96,21 +96,22 @@ class CategoriesHolder
...
-      {"ro", 33},
-      {"ru", 34},
-      {"sk", 35},
-      {"sv", 36},
-      {"sw", 37},
-      {"th", 38},
-      {"tr", 39},
-      {"uk", 40},
-      {"vi", 41},
+      {"lv", 27},
```
Comment (pastk) [inline indexer/categories_holder.hpp:99]: does it have to be in alphabetical order? (looks like no as `en` goes first)

does the ids change break compatibility with older map files?

Comment (biodranik) [reply indexer/categories_holder.hpp:99]: No specific order requirements. Alphabetic sorting is used for readers' comfort.

File: data/categories.txt [hunk 20]
```diff
@@ -16732,6 +17182,7 @@ et:Jooksurada
 fr:Piste de course
 it:Pista da corsa
 ja:トラック
+lv:Skrējceļš|Skriešanas celiņš
```
Comment (soshial) [inline data/categories.txt:17185]: Please fix to

`lv:Skrejceļš|Skriešanas celiņš`

File: data/categories.txt [hunk 21]
```diff
@@ -14185,6 +14563,7 @@ id:Kolam Renang
 it:Centro natatorio
 ja:スイミングセンター
 ko:수영 센터|수영장
+lv:Peldēšana
```
Comment (soshial) [inline data/categories.txt:14566]: Please add "Peldbaseins|Baseins|Peldvieta"

**Comment by soshial** [comment]
@biodranik thanks a lot for your effort of merging our corrections. I have left a couple more that were missed — after these are resolved we may merge this PR

File: indexer/categories_holder.hpp (hunk 19 above)
Comment (pastk) [reply indexer/categories_holder.hpp:99]: @vng why do we have hardcoded ids here if they don't matter? is it possible to change it to just provide a list of supported langs without ids?

**Review by pastk — APPROVED** [review]

**Comment by biodranik** [comment]
Updated, @vng PTAL

File: indexer/categories_holder.hpp (hunk 19 above)
Comment (vng) [reply indexer/categories_holder.hpp:99]: Good question. Probably, the initial idea was to make category lang codes the same as StringUtf8Multilang lang codes ..

**Review by vng — COMMENTED** [review]
Last commit is strange.

**Review by vng — APPROVED** [review]
Probable "Enable iOS" commit should follow before "Regenerated", but not a big deal.

**Comment by biodranik** [comment]
Merging it with the hope to facilitate types_strings.txt and other missing strings translation to Latvian @organicmaps/translations-lv

**Pull request merged**